packages/test-commerce-sdk-react/app/components/_app-config/index.tsx

```diff
@@ -73,12 +73,17 @@ AppConfig.restore = () => {}
 AppConfig.extraGetPropsArgs = () => {}
 AppConfig.freeze = () => {}
 
-// Recommended settings for PWA-Kit usage.
+const isServerSide = typeof window === 'undefined'
+// Recommended settings for PWA-Kit usages.
 const options = {
     queryClientConfig: {
         defaultOptions: {
             queries: {
-                retry: false
+                retry: false,
+                // Avoid double requests during hydration
+                staleTime: 5 * 1000,
```

Comment:
Do we have a reason that justifies the 5 seconds time period we consider all request fresh?

Reply:
Did we confirm if using the default value of 5 seconds will not cause to show not real data to users when they choose to apply optimistically UI updates?

Example. Users may want to seed the React Query cache to show instant transitions to the next component (PLP pagination, transition from PLP to PDP, etc.), but after the instant transition occurs with seeded data, they may want to present to the user real data. 
I believe this setting could potentially prevent users from seeing real data.

Reply:
I think your example here is a little contradictory @adamraya. If we choose a stale time, we choose a stale time. That value will be applied across the board. What makes the landscape more confusing is is we say something like, "you data is considered valid for X seconds, unless it's data that was pre-seeded, then that data is immediately considered stale"


An aside: 

If you did want to have a situation like that, could you not pre-seed data with a timestamp that is in the past. So the pre-seeded data's timestamp is "Date.now() - staleTime". So when that data is loaded it's immediately considered stale and refetched.

Reply:
That being said. I'd prefer to keep momentum up, by picking a stale time and put a note beside it saying you might have to customize this if you see want to keep data valid for longer or shorter.

Reply:
@adamraya No, I didn't verify the 5 seconds.. it's an arbitrary number.

You got a point there. I guess that's also one reason why the default value is 0 sec.

One advantage with the previous PR (#767 ) is that it doesn't touch the staleTime value. It uses `refetchOnMount: false` instead. But I don't want to use that option over here in this PR because it will be applied _all the time_. My previous PR set that option only briefly: after hydration, it will be reset to react-query's default.